### PR TITLE
use GitHub actions (linux support)

### PR DESCRIPTION
**Pull request by v1v**
### What

- Use GitHub actions for the CI
- Report test results
- Export GH action data to OTEL
- Use PreCommit GH actions

### Follow up

- Support windows
- Release automation

### CODE REVIEW

File: .github/workflows/test-reporter.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+## Workflow to process the JUnit test results and add a report to the checks.
+name: test-reporter
+on:
+  workflow_run:
+    workflows:
+      - ci
```
Comment (v1v) [inline .github/workflows/test-reporter.yml:7]:
```suggestion
      - test
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  test-linux:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Run release -c false within the docker container
+      run: |
+        docker run --rm -ti \
```
Comment (v1v) [inline .github/workflows/test.yml:53]:
```suggestion
        docker run --rm -t \
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -38,87 +38,37 @@ pipeline {
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
       }
     }
-    stage('Sanity checks') {
+    stage('Windows') {
```
Comment (v1v) [inline .ci/Jenkinsfile:41]: Changes are pretty much:
- https://github.com/elastic/ecs-dotnet/pull/236/commits/7cc57d9f35b7c92c4fec4dd025412e686468a261
- https://github.com/elastic/ecs-dotnet/pull/236/commits/607e5402eb79c47668a751c36d3661e4233d0f50
- https://github.com/elastic/ecs-dotnet/pull/236/commits/0412f5c0b632a860dded6dc724155d706f55784d

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+env:
+  # This is the version to be used for both OS platforms (windows and linux)
+  DOTNET_VERSION: '6.0.400'
+
+jobs:
+# TODO: There are some failures when running the tests
+#  - Failed to retrieve shard stats from node
+#  - Logs were not written to Elasticsearch within margin of 10 seconds
+# To be solved in a follow up
```
Comment (v1v) [inline .github/workflows/test.yml:30]: There are some errors in the ITs when running `build.bat release -c false`

- https://github.com/elastic/ecs-dotnet/actions/runs/4014234600/jobs/6894506499#step:5:1952

```
Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - Elastic.CommonSchema.Serilog.Sinks.IntegrationTests.dll (net6.0)
:: test: Failed! Process exited with '1' "dotnet "test" "-c" "RELEASE" "-m:1" "--logger:"junit;LogFilePath=D:\a\ecs-dotnet\ecs-dotnet\build\output\junit-{assembly}-{framework}-test-results.xml""" 
```


```
[2023-01-26T10:29:10,775][INFO ][o.e.c.m.MetadataCreateDataStreamService] [logging-node-c557469201] adding data stream [logs-1e4e11-2023.01.26] with write index [.ds-logs-1e4e11-2023.01.26-2023.01.26-000001], backing indices [], and aliases []
[2023-01-26T10:29:10,791][INFO ][o.e.c.m.MetadataCreateIndexService] [logging-node-c557469201] [x-dotnet-a7116d] creating index, cause [auto(bulk api)], templates [], shards [1]/[1]
  Failed Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToIndexTests.SerializesAndDeserializesMessageTemplateAndScope [19 s]
  Error Message:
   System.Exception : Logs were not written to Elasticsearch within margin of 10 seconds
  Stack Trace:
     at Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToIndexTests.SerializesAndDeserializesMessageTemplateAndScope() in D:\a\ecs-dotnet\ecs-dotnet\tests\Elasticsearch.Extensions.Logging.IntegrationTests\LoggingToIndexTests.cs:line 60
--- End of stack trace from previous location ---
  Failed Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToDataStreamTests.SerializesAndDeserializesMessageTemplateAndScope [18 s]
  Error Message:
   System.Exception : Logs were not written to Elasticsearch within margin of 10 seconds
  Stack Trace:
     at Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToDataStreamTests.SerializesAndDeserializesMessageTemplateAndScope() in D:\a\ecs-dotnet\ecs-dotnet\tests\Elasticsearch.Extensions.Logging.IntegrationTests\LoggingToDataStreamTests.cs:line 66
--- End of stack trace from previous location ---
[2023-01-26T10:29:19,999][INFO ][o.e.c.m.MetadataMappingService] [logging-node-c557469201] [.ds-logs-60abea-2023.01.26-2023.01.26-000001/DJkaaQteQm-P5uZRUB2nfA] update_mapping [_doc]
[2023-01-26T10:29:20,015][INFO ][o.e.c.m.MetadataMappingService] [logging-node-c557469201] [x-dotnet-a3d4a6/oCNRSXz1TcmXxk5r-oxj_Q] create_mapping
[xUnit.net 00:01:17.03]     Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToDataStreamTests.LogsEndUpInCluster [FAIL]
[xUnit.net 00:01:17.03]     Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToIndexTests.LogsEndUpInCluster [FAIL]
  Failed Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToDataStreamTests.LogsEndUpInCluster [10 s]
  Error Message:
   System.Exception : Logs were not written to Elasticsearch within margin of 10 seconds
  Stack Trace:
     at Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToDataStreamTests.LogsEndUpInCluster() in D:\a\ecs-dotnet\ecs-dotnet\tests\Elasticsearch.Extensions.Logging.IntegrationTests\LoggingToDataStreamTests.cs:line 36
--- End of stack trace from previous location ---
  Failed Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToIndexTests.LogsEndUpInCluster [10 s]
  Error Message:
   System.Exception : Logs were not written to Elasticsearch within margin of 10 seconds
  Stack Trace:
     at Elasticsearch.Extensions.Logging.IntegrationTests.LoggingToIndexTests.LogsEndUpInCluster() in D:\a\ecs-dotnet\ecs-dotnet\tests\Elasticsearch.Extensions.Logging.IntegrationTests\LoggingToIndexTests.cs:line 34
--- End of stack trace from previous location ---
Calling4692
java
success: 0
```

Hence, I commented out the current implementation, so the windows support will happen in a follow up, to unblock this PR for now

**Review by reakaleek — APPROVED** [review]

**Comment by Mpdreamz** [comment]
The windows failures are odd. Double checked these in the base library where they run succesfully: 

https://github.com/elastic/elastic-ingest-dotnet/pull/16

Will open a PR to work on reenabling windows in CI.

**Pull request merged**